scripts/globals/crafting.lua

```diff
@@ -115,21 +114,55 @@ function getTestItem(player, npc, craftID)
 
     switch (npc:getName()): caseof
     {
-        ["Abd-al-Raziq"] = function (x)     TItemList = TI_Alchemy end,
-        ["Peshi_Yohnts"] = function (x)     TItemList = TI_Bonecraft end,
-        ["Ponono"] = function (x)             TItemList = TI_Clothcraft end,
-        ["Piketo-Puketo"] = function (x)     TItemList = TI_Cooking end,
-        ["Thubu_Parohren"] = function (x)     TItemList = TI_Fishing end,
-        ["Reinberta"] = function (x)         TItemList = TI_Goldsmithing end,
-        ["Faulpie"] = function (x)             TItemList = TI_Leathercraft end,
-        ["Mevreauche"] = function (x)         TItemList = TI_Smithing end,
-        ["Ghemp"] = function (x)             TItemList = TI_Smithing end,
-        ["Cheupirudaux"] = function (x)     TItemList = TI_Woodworking end,
+        ["Abd-al-Raziq"] =      function (x)    TItemList = TI_Alchemy end,
+        ["Peshi_Yohnts"] =      function (x)    TItemList = TI_Bonecraft end,
+        ["Ponono"] =            function (x)    TItemList = TI_Clothcraft end,
+        ["Piketo-Puketo"] =     function (x)    TItemList = TI_Cooking end,
+        ["Thubu_Parohren"] =    function (x)    TItemList = TI_Fishing end,
+        ["Reinberta"] =         function (x)    TItemList = TI_Goldsmithing end,
+        ["Faulpie"] =           function (x)    TItemList = TI_Leathercraft end,
+        ["Mevreauche"] =        function (x)    TItemList = TI_Smithing end,
+        ["Ghemp"] =             function (x)    TItemList = TI_Smithing end,
+        ["Cheupirudaux"] =      function (x)    TItemList = TI_Woodworking end,
     }
 
     return TItemList[NextRank]
 end
 
+
+-----------------------------------
+-- getRankKeyItem by craftID
+-----------------------------------
+
+function getRankKeyItem(player,craftID)
+
+    local RankKI = 0
+
+    if craftID == 48 then -- fishing
```

Comment:
could this be a table instead of if/else?

Reply:
Removed the if/elseif stack replaced with ENUM (which is all this was doing anyway).

Reply:
clicky resolvy on conversationy~